mkosi.md

```diff
@@ -1200,136 +1207,143 @@ local directory:
-  before invoking the build script, after the `mkosi.postinst` script
-  is invoked.  It is called the second time with the "final" command
-  line parameter as the last step before the image is considered
-  complete. The environment variable `$BUILDROOT` points to the root
-  directory of the installation image. Additional verbs may be added
-  in the future, the script should be prepared for that. This script
-  may be used to alter the images without any restrictions, after all
-  software packages and built sources have been installed. This script
-  is more flexible than `mkosi.postinst` in two regards: it has access
-  to the host file system so it's easier to copy in additional files
-  or to modify the image based on external configuration, and the
-  script is run in the host, so it can be used even without emulation
-  even if the image has a foreign architecture.
-
-* `mkosi.mksquashfs-tool` may be an executable script. If it exists is
-  is called instead of `mksquashfs`.
-
-* `mkosi.nspawn` may be an nspawn settings file. If this exists it
-  will be copied into the same place as the output image file. This is
-  useful since nspawn looks for settings files next to image files it
-  boots, for additional container runtime settings.
-
-* `mkosi.cache/` may be a directory. If so, it is automatically used as
-  package download cache, in order to speed repeated runs of the tool.
-
-* `mkosi.builddir/` may be a directory. If so, it is automatically
+  via `.gitignore` entries is to use the `mkosi.output/` directory
+  (see below), which is an easy way to exclude all build artifacts.
+
+  The `$MKOSI_DEFAULT` environment variable will be set inside of this
+  script so that you know which `mkosi.default` (if any) was passed
+  in.
+
+* The **`mkosi.prepare`** script is invoked directly after the
+  software packages are installed, from within the image context. It
+  is once called for the *development* image (if this is enabled, see
+  above) with the "build" command line parameter, right before copying
+  the extra tree. It is called a second time for the *final* image
+  with the "final" command line parameter. This script has network
+  access and may be used to install packages from other sources than
+  the distro's package manager (e.g. `pip`, `npm`, ...), after all
+  software packages are installed but before the image is cached (if
+  incremental mode is enabled). This script is executed within
+  `$SRCDIR`. In contrast to a general purpose installation, it is safe
+  to install packages to the system (`pip install`, `npm install -g`)
+  instead of in `$SRCDIR` itself because the build image is only used
+  for a single project and can easily be thrown away and rebuilt so
+  there's no risk of conflicting dependencies and no risk of polluting
+  the host system.
+
+* The **`mkosi.postinst`** script is invoked as the penultimate step
+  of preparing an image, from within the image context. It is called
```

Comment:
```suggestion
  of preparing an image, from within the image context, if it exists. It is called
```
even though "may" is used further down in the paragraph, I think it's clearer to keep it up front.